device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttDeviceMethod.java

```diff
@@ -109,12 +109,12 @@ public void send(final IotHubTransportMessage message) throws TransportException
 
                 if (requestMap.containsKey(message.getRequestId()))
                 {
-                    switch (requestMap.remove(message.getRequestId()))
+                    if (requestMap.remove(message.getRequestId()) == DeviceOperations.DEVICE_OPERATION_METHOD_RECEIVE_REQUEST)
```

Comment:
replace block with 
```
if (requestMap.remove(message.getRequestId()) != DeviceOperations.DEVICE_OPERATION_METHOD_RECEIVE_REQUEST)
{
  throwMethodsTransportException("Mismatched request and response operation");
}
```

Reply:
Interesting that Intellij didn't do that for me. I'll fix it, sure